asn1-core/src/types/bit_string.rs

```diff
@@ -1,33 +1,120 @@
+/// image G3FacsimilePage ::= '100110100100001110110'B
+/// trailer BIT STRING ::= '0123456789ABCDEF'H
+/// body1 G3FacsimilePage ::= '1101'B
+/// body2 G3FacsimilePage ::= '1101000'B
+/// ```
+/// **Note** that `body1` and `body2` are distinct abstract values because trailing
+/// 0 bits are significant (due to there being no "NamedBitList" in the
+/// definition of G3FacsimilePage).
+/// # Example
+/// Use a bit string type with a size constraint to model the values of a fixed
+/// sized bit field.
+/// ```asn1
+/// BitField ::= BIT STRING (SIZE (12))
+/// map1 BitField ::= '100110100100'B
+/// map2 BitField ::= '9A4'H
+/// map3 BitField ::= '1001101001'B -- Illegal - violates size constraint.
+/// ```
+/// # Example
+/// Use a bit string type to model the values of a bit map, an ordered
+/// collection of logical variables indicating whether a particular condition
+/// holds for each of a correspondingly ordered collection of objects.
+/// ```asn1
+/// DaysOfTheWeek ::= BIT STRING { sunday(0), monday (1), tuesday(2),
+///     wednesday(3), thursday(4), friday(5), saturday(6) } (SIZE (0..7))
+///
+/// sunnyDaysLastWeek1 DaysOfTheWeek ::= {sunday, monday, wednesday}
+/// sunnyDaysLastWeek2 DaysOfTheWeek ::= '1101'B
+/// sunnyDaysLastWeek3 DaysOfTheWeek ::= '1101000'B
+/// sunnyDaysLastWeek4 DaysOfTheWeek ::= '11010000'B -- Illegal
+/// ```
+/// **Note** that if the bit string value is less than 7 bits long, then the
+/// missing bits indicate a cloudy day for those days, hence the first three
+/// values above have the same abstract value.
+/// # Example
+/// Use a bit string type to model the values of a bit map, a fixed-size ordered
+/// collection of logical variables indicating whether a particular condition
+/// holds for each of a correspondingly ordered collection of objects.
+/// ```asn1
+/// DaysOfTheWeek ::= BIT STRING { sunday(0), monday (1), tuesday(2),
+///     wednesday(3), thursday(4), friday(5), saturday(6) } (SIZE (7))
+///
+/// sunnyDaysLastWeek1 DaysOfTheWeek ::= {sunday, monday, wednesday}
+/// sunnyDaysLastWeek2 DaysOfTheWeek ::= '1101'B -- Illegal -- violates size constraint.
+/// sunnyDaysLastWeek3 DaysOfTheWeek ::= '1101000'B
+/// sunnyDaysLastWeek4 DaysOfTheWeek ::= '11010000'B -- Illegal -- violates size constraint.
+/// ```
+/// **Note** that the first and third values have the same abstract value.
+/// # Example
+/// Use a bit string type with named bits to model the values of a collection of
+/// related logical variables.
+/// ```asn1
+/// PersonalStatus ::= BIT STRING { married(0), employed(1), veteran(2), collegeGraduate(3) }
+/// jane PersonalStatus ::= { married, employed, collegeGraduate }
+/// alice PersonalStatus ::= '110100'B
+/// ```
+/// **Note** that `jane` and `alice` have the same abstract values.
 #[derive(Clone, Debug, Default, Deserialize, PartialEq)]
 #[serde(rename="ASN.1#BitString")]
 pub struct BitString(BitVec);
 
 impl BitString {
+    /// Instantiate a new empty instance of `BitString`.
```

Comment:
This could also be simplified to say
```suggestion
    /// Creates an empty `BitString`.
```
Or if not, I think it would be better to at least include an `s` like so:
```suggestion
    /// Instantiates a new empty instance of `BitString`.
```